Library/Homebrew/cmd/pull.rb

```diff
@@ -126,42 +120,58 @@ def pull
         end
       end
 
+      orig_message = message = `git log HEAD^.. --format=%B`
       if issue && !ARGV.include?("--clean")
         ohai "Patch closes issue ##{issue}"
-        message = `git log HEAD^.. --format=%B`
+        # If this is a pull request, append a close message.
+        unless message.include? "Closes ##{issue}."
+          message += "\nCloses ##{issue}."
+        end
+      end
 
-        if ARGV.include? "--bump"
-          odie "Can only bump one changed formula" unless changed_formulae.length == 1
-          formula = changed_formulae.first
-          subject = "#{formula.name} #{formula.version}"
+      if changed_formulae.empty?
+        odie "cannot bump: no changed formulae found after applying patch" if do_bump
+        is_bumpable = false
+      end
+      if is_bumpable && !ARGV.include?("--clean")
+        formula = changed_formulae.first
+        new_versions = {
+            stable: formula.stable.nil? ? nil : formula.stable.version.to_s,
+            devel: formula.devel.nil? ? nil : formula.devel.version.to_s
```

Comment:
I don't think we use this hash syntax anywhere else and it's incompatible with Ruby 1.8 (though that is less of a concern in this particular command). This should be:

```ruby
new_versions = {
  :stable => formula.stable.nil? ? nil : formula.stable.version.to_s,
  :devel => formula.devel.nil? ? nil : formula.devel.version.to_s,
}
```